keps/sig-cli/2299-kustomize-plugin-composition/README.md

```diff
@@ -185,62 +210,55 @@ modules:
 
-### Risks and Mitigations
-
-#### Security of running containers locally
-
-Composition has similar capabilities to Kustomization.
+#### Integration with Kubectl Kustomize
 
-The key difference for security is that it does not require plugins to be installed in advance / out of band. This difference is
-key to the user experience, and it does decrease user awareness of exactly what is
-being executed. As such, it must be made clear to users that Compositions are not inherently
-secure and that they should only build Compositions they trust.
+`Composition` support will neither be required nor suppressed in `kubectl kustomize` during alpha. In
+other words, if a `kubectl kustomize` release happens during alpha, it can include the feature, but
+graduation to beta does not require this. `Composition` support MUST be included in at least two
+`kubectl kustomize` releases during beta.
 
-This risk will be mitigated in the following ways:
+#### Plugin execution flags
 
-1. Container-based functions will be run without network or volume access. This does
-limit the functionality they can implement, but it also encourages plugins
-to be developed such that the module resources that configure them express
-the entire desired state. Should this prove unacceptably limiting, network or volume access
-may be introduced in a future version, but likely with a more explicit and restrictive
-permissions model than plugins have today. The same is true for exec plugins: they
-will not be initially supported, and additional restrictions will likely be placed on them
-if support is introduced in the future.
+This KEP does not propose any changes to the flags used to gate plugins, or to the parameters
+available to configure container/exec/starlark plugin execution. All plugin execution from either
+Kind will remain gated behind `--enable-alpha-plugins`, as well as existing additional flags for
+particular plugin provider types. Any changes to plugin gating should apply identically to both
+Kinds. This does limit the usefulness of `Composition` within `kubectl -k` for the time being, since
+only a subset of plugins can be enabled in that context.
 
-1. Even if network or volume access support is added, containers will still run
-without it unless the user's Composition explicitly grants it. If exec support
-is added, a checksum may be required in addition to the binary location.
+### Risks and Mitigations
 
-1. Plugins will not be trusted by default. The execution framework will feature
-a pluggable trust model that enables users and organizations to allowlist plugin
-sources. It should be possible for large organizations to distribute binaries with
-specific allowlists compiled in, and for additional sources to be allowlisted at
-invocation time.
+This proposal effectively takes a feature set that exists in Kustomize today and makes it much more
+usable in certain workflows. As such, it does not come with any novel technical risks.
```

Comment:
I noticed you removed the section stating that container-based functions will be run without networking or volume access. Does that mean Composition will support the `network` and `mounts` fields? 

This question is coming in part from wanting clarity about the syntax proposed in https://github.com/kubernetes/enhancements/pull/2908/files#r699656279, which seems to suggest changing the container spec; I'm wondering if Composition is assuming the same changed UX.

Reply:
Correct, this KEP itself doesn't change the fields required/supported for any particular plugin type. It will support whatever options are present when it merges.